### PR TITLE
Update default etcd server to 3.4.3

### LINKED ISSUE #82641: Update to etcd 3.4 in Kubernetes 1.17 release
follow the same pattern that we updated to 3.3.15:
- client side changes needs to be vendored in (similar to https://github.com/kubernetes/kubernetes/pull/81434)
- update default etcd server (similar to https://github.com/kubernetes/kubernetes/pull/82199)

### PR DESCRIPTION
**Pull request by jingyih**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
We are updating etcd client version to 3.4.3 in #83987. This PR updates the default etcd server version to 3.4.3.

**Does this PR introduce a user-facing change?**:
```release-note
Update default etcd server version to 3.4.3
```

/sig api-machinery
/area etcd
/priority important-soon

Fixes: #82641

### CODE REVIEW

**Comment by jingyih** [comment]
/hold
Until 3.4.3 image is pushed.
Until #83987 is merged.

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -425,8 +425,8 @@ var (
 		14: "3.3.10",
 		15: "3.3.10",
 		16: "3.3.15-0",
-		17: "3.3.17-0",
```
Comment (wenjiaswe) [inline cmd/kubeadm/app/constants/constants.go]: Do we want to change default of 16 to 3.3.17-0?

Comment (jingyih) [reply cmd/kubeadm/app/constants/constants.go]: Make sense. ref: #83968

Comment (jpbetz) [reply cmd/kubeadm/app/constants/constants.go]: I'm in favor of updating to same patch version for all k8s versions using 3.3 in kubeadm.

**Comment by BenTheElder** [comment]
what's the process to get the images pushed?

**Comment by jingyih** [comment]
> what's the process to get the images pushed?

My current understanding of the process:

1. build and push etcd-related images to `gcr.io/k8s-image-staging`

2. test the new images in K8s cluster

3. promote images from staging repo to `gcr.io/google-containers` via registry promoter.

**Comment by jpbetz** [comment]
cc @listx for ongoing `gcr.io/google-containers` promotion improvement efforts

**Comment by neolit123** [comment]
/approve
for kubeadm. thanks!

**Comment by jingyih** [comment]
The 3.4.3 version of `etcd` and `etcd-empty-dir-cleanup` images are available in `gcr.io/google-containers`.
Can we get an "ok-to-test"?

**Comment by jingyih** [comment]
All tests passed. PTAL.

/hold cancel

**Comment by neolit123** [comment]
/test pull-kubernetes-e2e-kind
should end in about 20 minutes (creates a kubeadm / kind cluster).

**Comment by wojtek-t** [comment]
/lgtm
/approve

Let's merge this asap to have as much soaking time as possible before the release.

**Pull request merged**